contracts/StakingContractHandler.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity 0.5.16;
+
+import "./ContractRegistryAccessor.sol";
+import "./spec_interfaces/IStakingContractHandler.sol";
+
+contract StakingContractHandler is IStakingContractHandler, ContractRegistryAccessor {
+
+    uint constant notificationGasLimit = 5000000;
+
+    modifier onlyStakingContract() {
+        require(msg.sender == address(getStakingContract()), "caller is not the staking contract");
+
+        _;
+    }
+
+    function stakeChange(address _stakeOwner, uint256 _amount, bool _sign, uint256 _updatedStake) external onlyStakingContract {
+        IDelegations delegationsContract = getDelegationsContract();
```

Comment:
It's better to use IStakeChangeNotifier (subset of IDelegations) to avoid locking the interface